transformer_engine/pytorch/csrc/extensions/gemm.cpp

```diff
@@ -251,23 +274,24 @@ std::vector<py::object> gemm(py::handle A, bool transa, py::handle B, bool trans
     } else {
       // Launch GEMM
       NVTE_SCOPED_GIL_RELEASE({
-        nvte_cublas_gemm_scaled(A_tensor.data(), B_tensor.data(), D_tensor.data(),
+        nvte_cublas_gemm_scaled(A_tensor.data(), B_tensor.data(), out_tensor.data(),
                                 bias_tensor.data(), te_pre_gelu_out.data(), transa, transb, grad,
                                 te_workspace.data(), alpha, *beta, use_split_accumulator,
                                 num_math_sms, main_stream);
       });
     }
   } else {
-    if (D_tensor.numel() != 0 && !accumulate) {
-      D_tensor.zero_(main_stream);
+    if (out_tensor.numel() != 0 && !accumulate) {
+      out_tensor.zero_(main_stream);
     }
     if (bias.has_value()) {
       if (bias->numel() != 0 && grad) {
         bias_grad->zero_();
       }
     }
   }
-
+  std::unique_ptr<Quantizer> my_quantizer = convert_quantizer(quantizer);
+  if (unfused_quantization_needed) my_quantizer->quantize(unquantized_D_tensor, D_tensor);
```

Comment:
It feels a bit neater only construct the C++ quantizer when needed, especially since the constructor calls Python:
```suggestion
  if (unfused_quantization_needed) {
    convert_quantizer(quantizer)->quantize(unquantized_D_tensor, D_tensor);
  }
```